### PR TITLE
Make local package sources (path) available for meta.yaml

### PR DESCRIPTION
**Pull request by phorward**
Hi there!

This issue should only provide as a discussion base for a feature I would like to request and how I resolved it for case. My problem is that I want to build a package for a tiny library called html5 which currently is not made available in pip, so no way to fetch it from a URL in the usual way (via source: url+checksum). 

So I created a meta.yaml like the following and modified pyodide to even build packages from local sources, by accepting a path parameter and copying contents from this directory into the build directory:
```
package:
  name: html5
  version: "2.4"

source:
  path: src

test:
  imports:
    - html5
```
This worked for me in some tests, maybe there's a better way to provide this.
If you like it or have improvements, feel free to add it, otherwise reject this.

Thanks for making this great stuff happen! :+1: 

**Edit** (by rth): also closes https://github.com/iodide-project/pyodide/issues/467

### CODE REVIEW

**Comment by mdboom** [comment]
Seems like a useful feature.  Can you add docs to `docs/new_packages.md`?

**Comment by phorward** [comment]
> Seems like a useful feature. Can you add docs to `docs/new_packages.md`?

Sure, I've recently added a section to new_packages.md!

File: docs/new_packages.md [hunk 1]
```diff
@@ -66,6 +66,14 @@ The url of the source tarball.
 The tarball may be in any of the formats supported by Python's
 `shutil.unpack_archive`: `tar`, `gztar`, `bztar`, `xztar`, and `zip`.
 
+#### `source/path`
+
+Alternatively to `source/url`, a relative or absolute path can be specified
+as package source. This is useful for local testing or building packages which
+are not available online in the required format.
+
+If a path is specified, any provided checksums are being ignored.
```
Comment (mdboom) [inline docs/new_packages.md]:
```suggestion
If a path is specified, any provided checksums are ignored.
```

File: pyodide_build/buildpkg.py [hunk 2]
```diff
@@ -40,15 +40,30 @@ def check_checksum(path, pkg):
...
+             subprocess.run([
+                  'wget', '-q', '-O', str(tarballpath), pkg['source']['url']
+             ], check=True)
+             check_checksum(tarballpath, pkg)
+
+        if not srcpath.is_dir():
+            shutil.unpack_archive(str(tarballpath), str(buildpath))
+
+    elif 'path' in pkg['source']:
+        srcdir = Path(pkg['source']['path'])
```
Comment (rth) [inline pyodide_build/buildpkg.py]: Is the relative path relative to the file or to the home directory?

**Comment by rth** [comment]
There are still some linting issues that need fixing as well.

File: pyodide_build/buildpkg.py (hunk 2 above)
Comment (phorward) [reply pyodide_build/buildpkg.py]: Hi, the path currently is relative to the package directory where the meta.yaml is located.

**Comment by phorward** [comment]
Hi, I've fixed the linting issues, but there still is one check "test-chrome" failing.

**Comment by mdboom** [comment]
Thanks.  The `test-chrome` failures are a known intermittent problem (unrelated to this PR).  I've fixed the merge conflicts, and will give testing another round.  Assuming that passes, I think this is good to go.  Thanks again!

**Comment by msabramo** [comment]
This looks like a potentially useful feature. 👍

**Comment by phorward** [comment]
@mdboom: I've recently updated the branch again, now all should be clean! I'm starting over with another branch in my repo to integrate my changes that should not come into upstream. Sorry again ;-)

**Comment by rth** [comment]
So I have restarted the CI twice in this PR and each time there is one of the following,
```
packages/scipy/test_scipy.py::test_scipy_linalg[firefox]
packages/scipy/test_scipy.py::test_brentq[firefox]
```
that fails (not always the same one). Though I think tests are just flaky and if there was an issue in this PR I would expect the failures to be more deterministic. Merging. We'll see how it goes on master.

Thank you @phorward !

**Pull request merged**